java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/DateTimeUtils.java

```diff
@@ -32,7 +33,7 @@ private DateTimeUtils() {
    */
   public static long applyCalendarOffset(long milliseconds, Calendar calendar) {
     if (calendar == null) {
-      return milliseconds;
+      return milliseconds - Calendar.getInstance(TimeZone.getDefault()).getTimeZone().getOffset(milliseconds);
```

Comment:
Shouldn't we use UTC by default? I feel like using the system timezone is wrong (There's even a ticket to remove this behavior in the legacy driver).

Reply:
this is behaviour we have in  PostgreSQL jdbc driver.  Dremio jdbc driver just throw on null calendar.